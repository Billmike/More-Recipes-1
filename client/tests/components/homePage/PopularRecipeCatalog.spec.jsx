import React from 'react';

import PopularRecipeCatalog from '../../../components/homepage/PopularRecipesCatalog'; // eslint-disable-line

describe('Popular recipe catalog component', () => {
  it('should render component', () => {
    const props = {
      recipes: {
        isFetchingPopularRecipes: false,
        popularRecipes: [],
      }
    };
    const wrapper = mount(<PopularRecipeCatalog {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with spinner when fetching recipes', () => {
    const props = {
      recipes: {
        isFetchingPopularRecipes: true,
        popularRecipes: [],
      }
    };
    const wrapper = mount(<PopularRecipeCatalog {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with child recipe catalog card', () => {
    const props = {
      recipes: {
        isFetchingPopularRecipes: false,
        popularRecipes: [{
          id: 1,
          title: 'Rice',
          category: 'Dinner',
          description: 'Tasty rice',
          preparationTime: 69,
          ingredients: 'Rice, tomatoes, salt',
          directions: 'Do this. Do this. Do that.',
          upvotes: 2,
          downvotes: 0,
          favorites: 5,
          votes: 3,
        }, {
          id: 2,
          title: 'Yam',
          category: 'Breakfast',
          description: 'Tasty yam',
          preparationTime: 50,
          ingredients: 'Yam, tomatoes, egg, salt',
          directions: 'Do this. Do this. Do that.',
          upvotes: 5,
          downvotes: 2,
          favorites: 4,
          votes: 5,
        }],
      },
    };
    const wrapper = shallow(<PopularRecipeCatalog {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('RecipeCatalogCard').length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });
});
