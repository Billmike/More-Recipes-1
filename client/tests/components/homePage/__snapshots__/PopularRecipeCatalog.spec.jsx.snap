// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular recipe catalog component should render component 1`] = `
<PopularRecipesCatalog
  recipes={
    Object {
      "isFetchingPopularRecipes": false,
      "popularRecipes": Array [],
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col s12 popular-recipes"
    >
      <h4>
        Popular Recipes
      </h4>
    </div>
    <div />
  </div>
</PopularRecipesCatalog>
`;

exports[`Popular recipe catalog component should render component with child recipe catalog card 1`] = `
<div
  className="row"
>
  <div
    className="col s12 popular-recipes"
  >
    <h4>
      Popular Recipes
    </h4>
  </div>
  <div>
    <RecipeCatalogCard
      index={0}
      key="1"
      recipe={
        Object {
          "category": "Dinner",
          "description": "Tasty rice",
          "directions": "Do this. Do this. Do that.",
          "downvotes": 0,
          "favorites": 5,
          "id": 1,
          "ingredients": "Rice, tomatoes, salt",
          "preparationTime": 69,
          "title": "Rice",
          "upvotes": 2,
          "votes": 3,
        }
      }
      recipes={
        Object {
          "isFetchingPopularRecipes": false,
          "popularRecipes": Array [
            Object {
              "category": "Dinner",
              "description": "Tasty rice",
              "directions": "Do this. Do this. Do that.",
              "downvotes": 0,
              "favorites": 5,
              "id": 1,
              "ingredients": "Rice, tomatoes, salt",
              "preparationTime": 69,
              "title": "Rice",
              "upvotes": 2,
              "votes": 3,
            },
            Object {
              "category": "Breakfast",
              "description": "Tasty yam",
              "directions": "Do this. Do this. Do that.",
              "downvotes": 2,
              "favorites": 4,
              "id": 2,
              "ingredients": "Yam, tomatoes, egg, salt",
              "preparationTime": 50,
              "title": "Yam",
              "upvotes": 5,
              "votes": 5,
            },
          ],
        }
      }
    />
    <RecipeCatalogCard
      index={1}
      key="2"
      recipe={
        Object {
          "category": "Breakfast",
          "description": "Tasty yam",
          "directions": "Do this. Do this. Do that.",
          "downvotes": 2,
          "favorites": 4,
          "id": 2,
          "ingredients": "Yam, tomatoes, egg, salt",
          "preparationTime": 50,
          "title": "Yam",
          "upvotes": 5,
          "votes": 5,
        }
      }
      recipes={
        Object {
          "isFetchingPopularRecipes": false,
          "popularRecipes": Array [
            Object {
              "category": "Dinner",
              "description": "Tasty rice",
              "directions": "Do this. Do this. Do that.",
              "downvotes": 0,
              "favorites": 5,
              "id": 1,
              "ingredients": "Rice, tomatoes, salt",
              "preparationTime": 69,
              "title": "Rice",
              "upvotes": 2,
              "votes": 3,
            },
            Object {
              "category": "Breakfast",
              "description": "Tasty yam",
              "directions": "Do this. Do this. Do that.",
              "downvotes": 2,
              "favorites": 4,
              "id": 2,
              "ingredients": "Yam, tomatoes, egg, salt",
              "preparationTime": 50,
              "title": "Yam",
              "upvotes": 5,
              "votes": 5,
            },
          ],
        }
      }
    />
  </div>
</div>
`;

exports[`Popular recipe catalog component should render component with spinner when fetching recipes 1`] = `
<PopularRecipesCatalog
  recipes={
    Object {
      "isFetchingPopularRecipes": true,
      "popularRecipes": Array [],
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col s12 popular-recipes"
    >
      <h4>
        Popular Recipes
      </h4>
    </div>
    <div
      className="center-align"
    >
      <Spinner
        size="big"
      >
        <div
          className="preloader-wrapper big active valign-wrapper"
        >
          <div
            className="spinner-layer spinner-blue"
          >
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="gap-patch"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="circle-clipper right"
            >
              <div
                className="circle"
              />
            </div>
          </div>
          <div
            className="spinner-layer spinner-red"
          >
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="gap-patch"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="circle-clipper right"
            >
              <div
                className="circle"
              />
            </div>
          </div>
          <div
            className="spinner-layer spinner-yellow"
          >
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="gap-patch"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="circle-clipper right"
            >
              <div
                className="circle"
              />
            </div>
          </div>
          <div
            className="spinner-layer spinner-green"
          >
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="gap-patch"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="circle-clipper right"
            >
              <div
                className="circle"
              />
            </div>
          </div>
        </div>
      </Spinner>
    </div>
  </div>
</PopularRecipesCatalog>
`;
